fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -314,6 +331,38 @@ public void close() {
         return ret;
     }
 
+    /**
+     * Run after a single transactional call within runAsync.
+     */
+    private <R> Pair<R, Throwable> runAsyncPostTransaction(R result, Throwable exception) {
+        if (exception == null) {
+            successCount++;
+            if (successCount >= increaseLimitAfter && limit < maxLimit) {
+                increaseLimit();
+            }
+        } else {
+            successCount = 0;
+        }
+        return Pair.of(result, exception);
+    }
+
+    private void decreaseLimit(FDBException fdbException) {
+        limit = Math.max(1, (3 * limit) / 4);
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(KeyValueLogMessage.of("Lessening limit of online index build",
+                    "indexName", index.getName(),
+                    "indexVersion", index.getLastModifiedVersion(),
+                    "error", fdbException.getMessage(),
+                    "errorCode", fdbException.getCode(),
+                    "limit", limit),
+                    fdbException);
+        }
+    }
+
+    private void increaseLimit() {
+        limit = Math.min(maxLimit, Math.max(limit + 1, (4 * limit) / 3));
```

Comment:
Do you think increasing the limit should also get a log message? Maybe at `DEBUG`?

Alternatively, maybe it's not necessary if instead the limit is included in the period "progress" message.